crates/driver/src/tests/cases/protocol_fees.rs

```diff
@@ -114,6 +145,38 @@ async fn surplus_protocol_fee_buy_order_not_capped() {
     protocol_fee_test_case(test_case).await;
 }
 
+#[tokio::test]
+#[ignore]
+async fn surplus_protocol_fee_buy_partial_order() {
+    let fee_policy = Policy::Surplus {
+        factor: 0.5,
+        // high enough so we don't get capped by volume fee
+        max_volume_factor: 1.0,
+    };
+    let test_case = TestCase {
+        fee_policy,
+        order: Order {
+            sell_amount: 50.ether().into_wei(),
+            buy_amount: 40.ether().into_wei(),
+            side: order::Side::Buy,
+        },
+        execution: Execution {
+            // 6 ETH surplus in sell token (after network fee), half of which is kept by the
+            // protocol
+            solver: Amounts {
+                sell: 41.ether().into_wei(),
+                buy: 28.ether().into_wei(),
+            },
+            driver: Amounts {
+                sell: 38.ether().into_wei(),
```

Comment:
Driver amount is 38 while the solver amount is 41. How is this possible? I would expect the driver sell execution amount to be >= solver execution amount?